lib/event-service.js

```diff
@@ -0,0 +1,59 @@
+var EVENTSERVICE_SERVICE_UUID          = 'E95D93AF251D470AA062FA1922DFA9A8'.toLowerCase();
+var MICROBITEVENT_CHARACTERISTIC_UUID = 'E95D9775251D470AA062FA1922DFA9A8'.toLowerCase();
+
+var CLIENTREQUIREMENTS_CHARACTERISTIC_UUID = 'E95D23C4251D470AA062FA1922DFA9A8'.toLowerCase();
+var CLIENTEVENT_CHARACTERISTIC_UUID = 'E95D5404251D470AA062FA1922DFA9A8'.toLowerCase();
+
+var microbit_event_subscribed = 0;
+
+var EventService = function() {
+};
+
+EventService.prototype.hasEventService = function() {
+  return this.hasService(EVENTSERVICE_SERVICE_UUID);
+};
+
+EventService.prototype.onMicrobitEventChange = function(data) {
+  if (data.length !== 4) {
+    return;
+  }
+  var event_id = data.readInt16LE(0);
+  var event_value = data.readInt16LE(2);
+  this.emit('microbitEventChange', event_id, event_value);
+};
+
+EventService.prototype.subscribeMicrobitEvents = function(client_req_event_id, client_req_event_value, callback) {
+  // specifying which events we want to be notified about
+  console.log("writing to micro:bit event requirements");
+  this.writeClientEventRequirements(client_req_event_id, client_req_event_value);
+
+  if (microbit_event_subscribed === 0) {
+    console.log("Subscribing for micro:bit event notifications");
+    this.onMicrobitEventChangeBinded = this.onMicrobitEventChange.bind(this);
+    this.subscribeCharacteristic(EVENTSERVICE_SERVICE_UUID, MICROBITEVENT_CHARACTERISTIC_UUID, this.onMicrobitEventChangeBinded, callback);
+    microbit_event_subscribed = 1;
+  }
+};
+
+EventService.prototype.unsubscribeMicrobitEvent = function(callback) {
+  // allow this event if not previously explicitly subscribed through a call to subscribeMicrobitEvents because the micro:bit may have persisted the 
+  // client characteristic configuration descriptor state from a previous "session"
+  this.unsubscribeCharacteristic(EVENTSERVICE_SERVICE_UUID, MICROBITEVENT_CHARACTERISTIC_UUID, this.onEventBinded, callback);
+  microbit_event_subscribed = 0;
+};
+
+EventService.prototype.writeClientEventRequirements = function(client_req_event_id, client_req_event_value, callback) {
+  var client_reqs = new Buffer(4);
+  client_reqs[0] = (client_req_event_id & 0xFF);

```

Comment:
I think this can be simplified with some of the Node.js Buffer API's: https://nodejs.org/api/buffer.html

like `buffer.writeUInt16Le(value, offset).